cmake/SSGCommon.cmake

```diff
@@ -94,20 +94,28 @@ macro(ssg_build_xccdf_oval_ocil PRODUCT)
     list(APPEND STIG_REFERENCE_FILE_LIST "not-found")
     list(GET STIG_REFERENCE_FILE_LIST 0 STIG_REFERENCE_FILE)
 
+    add_custom_command(
+        OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/product.yml"
+        COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${SSG_BUILD_SCRIPTS}/compile_product.py" --product-yaml "${CMAKE_SOURCE_DIR}/products/${PRODUCT}/product.yml" --compiled-product-yaml "${CMAKE_CURRENT_BINARY_DIR}/product.yml"
+        COMMENT "[${PRODUCT}-content] compiling product yaml"
+    )
+
```

Comment:
`"${CMAKE_CURRENT_BINARY_DIR}/product.yml"` is a necessary to build the `"${BUILD_CHECKS_DIR}/sce/metadata.json"`. So maybe this should be moved elsewhere?

Reply:
I have put it right next to compile everything, because it is very close to what that command does. Do you have any specific suggestions?

Reply:
Since it is an early artifact required for almost everything, I would tend to put it directly in the `ssg_build_product` macro, before the call to `ssg_build_sce` macro.

But I don't have a strong position on this, as I see that things might still change.

Reply:
This is easy to change later, so I would suggest to rather merge if it is not an obstacle.